### PR TITLE
[PyOV] Migrate from direct setup.py calls in Benchmark App and Docs CI

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - Remove direct `setup.py` calls from comments in cmake files
 - Change the docs building command in GHA
 - Migrate Benchmark App wheel building command to use `python -m pip wheel .`
 - Since `openvino_dev` is deprecated and set to be removed soon anyway, the `constraints.txt` file has been unpinned from Benchmark App requirements. In my opinion this simplifies the code significantly with no drawbacks, because `numpy` version for benchmark_app (the only dependency) is not even specified in `constraints.txt`, which will soon be removed. 

### Tickets:
 - CVS-155956
 - CVS-155957

### CODE REVIEW

**Review by Wovchena — APPROVED** [review]
Can we just ignore the need for `build`? Alternatively, one can use `pip wheel . --no-deps` to build it.

File: docs/openvino_sphinx_theme/README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 1. Install the `openvino_sphinx_theme` using `python`:
 ```
-python setup.py install --user
+python -m pip install --user
```
Comment (ilya-lavrenov) [inline docs/openvino_sphinx_theme/README.md:7]: you need to specify folder

File: tools/benchmark_tool/setup.py [hunk 2]
```diff
@@ -6,92 +6,18 @@
 """
 Use this script to create a wheel with OpenVINO™ Python* tools:
 
-$ python setup.py sdist bdist_wheel
+$ python -m build
```
Comment (ilya-lavrenov) [inline tools/benchmark_tool/setup.py:9]: folder?

File: tools/benchmark_tool/requirements.txt [hunk 3]
```diff
@@ -1,2 +1,2 @@
--c ../constraints.txt
 numpy>=1.16.6,<2.1.0
+build<1.3
```
Comment (ilya-lavrenov) [inline tools/benchmark_tool/requirements.txt:2]: I suppose `requirements` are supposed to be `deployment`, but `build` ones

**Review by ilya-lavrenov — CHANGES_REQUESTED** [review]

File: docs/openvino_sphinx_theme/README.md (hunk 1 above)
Comment (p-wysocki) [reply docs/openvino_sphinx_theme/README.md]: Done.

File: tools/benchmark_tool/setup.py (hunk 2 above)
Comment (p-wysocki) [reply tools/benchmark_tool/setup.py]: Changed to `python -m pip wheel .` as suggested in other comment.

File: tools/benchmark_tool/requirements.txt (hunk 3 above)
Comment (p-wysocki) [reply tools/benchmark_tool/requirements.txt]: Removed `build` from requirements, should no longer be an issue.

**Comment by p-wysocki** [comment]
> Can we just ignore the need for build? Alternatively, one can use pip wheel . --no-deps to build it.

I removed `build` and instructed to use `python -m pip wheel .` instead. I did not add `--no-deps` since `build` now can be removed from `requirements.txt`.

**Review by akuporos — APPROVED** [review]

File: docs/documentation_build_instructions.md [hunk 4]
```diff
@@ -26,7 +26,7 @@ $ source env/bin/activate
 ```
 5. Install the sphinx theme
 ```
-(env) $ cd docs/openvino_sphinx_theme && python setup.py install && cd -
+(env) $ cd docs/openvino_sphinx_theme && python -m pip install && cd -
```
Comment (ilya-lavrenov) [inline docs/documentation_build_instructions.md:29]:
```suggestion
(env) $ python -m pip install docs/openvino_sphinx_theme
```

**Review by almilosz — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

File: tools/benchmark_tool/setup.py [hunk 5]
```diff
@@ -6,92 +6,18 @@
...
+$ python -m pip wheel .
 """
-import pkg_resources
-import re
 from setuptools import setup, find_packages
-from pathlib import Path
-from typing import Dict, List
 
 
+with open('requirements.txt', 'r', encoding='utf-8') as f:
```
Comment (mryzhov) [inline tools/benchmark_tool/setup.py:14]: looks like we don't need setup.py here anymore, I would recommend to start using pyproject.toml and drop setup.py at all.

**Review by mryzhov — APPROVED** [review]

File: tools/benchmark_tool/setup.py (hunk 5 above)
Comment (p-wysocki) [reply tools/benchmark_tool/setup.py:14]: Already planned in CVS-156008.

**Pull request merged**